### PR TITLE
Fix VK_EXT_pipeline_robustness to check if exposed

### PR DESCRIPTION
**Pull request by spencer-lunarg**
To help, but not fix yet, https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/5657

Enabling robustness features can cause performance on GPUs to go down, the `VK_EXT_pipeline_robustness` extension was added so people could just use that extension, but it took advantage of the other robustness features

This change

- Moves logic from Stateless to CoreCheck
- creates a `has_robust_image_access` to check if extension is expose
- Disables the 4 `VK_EXT_robustness2` VUs
  - We will need to write up a way to query the device features if they are not provided, currently we just assumed the features were passed into CreateDevice

### CODE REVIEW

File: layers/core_checks/pipeline_cc_validation.cpp [hunk 1]
```diff
@@ -254,6 +254,13 @@ bool CoreChecks::ValidatePipeline(const PIPELINE_STATE &pipeline) const {
                              pipeline.create_index, attachment_sample_count_info->depthStencilAttachmentSamples);
         }
     }
+
+    if (const auto *pipeline_robustness_info = LvlFindInChain<VkPipelineRobustnessCreateInfoEXT>(pipeline.PNext());
+        pipeline_robustness_info) {
+        std::stringstream parameter_name;
+        parameter_name << "vkCreateGraphicsPipelines() pCreateInfos[" << pipeline.create_index << "]";
```
Comment (arno-lunarg) [inline layers/core_checks/pipeline_cc_validation.cpp:261]:
```suggestion
        parameter_name << "vkCreateGraphicsPipelines(): pCreateInfos[" << pipeline.create_index << "]";
```

File: layers/core_checks/pipeline_cc_validation.cpp [hunk 2]
```diff
@@ -1096,6 +1158,13 @@ bool CoreChecks::ValidateRayTracingPipeline(const PIPELINE_STATE &pipeline,
         skip |= ValidatePipelineShaderStage(pipeline, stage_state);
     }
 
+    if (const auto *pipeline_robustness_info = LvlFindInChain<VkPipelineRobustnessCreateInfoEXT>(create_info.pNext);
+        pipeline_robustness_info) {
+        std::stringstream parameter_name;
+        parameter_name << "vkCreateRayTracingPipelinesKHR: pCreateInfos[" << pipeline.create_index << "]";
```
Comment (arno-lunarg) [inline layers/core_checks/pipeline_cc_validation.cpp:1164]:
```suggestion
        parameter_name << "vkCreateRayTracingPipelinesKHR(): pCreateInfos[" << pipeline.create_index << "]";
```

**Review by arno-lunarg — APPROVED** [review]
LGTM! 2 suggestions to error strings

File: layers/core_checks/pipeline_cc_validation.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (spencer-lunarg) [reply layers/core_checks/pipeline_cc_validation.cpp:261]: I agree, honestly near-future task is to improve this all over pipeline validation as I would love a proper error object to pass around

My goal was to start https://github.com/orgs/KhronosGroup/projects/12?pane=issue&itemId=25404491 with this pipeline soon

**Review by artem-lunarg — APPROVED** [review]

**Pull request merged**